cdn-server/cdn/src/index.ts

```diff
@@ -88,13 +89,57 @@ const persistedOperation = (storage: BlobStorage) => {
   };
 };
 
+const routerConfig = (storage: BlobStorage) => {
+  return async (c: Context) => {
+    const organizationId = c.req.param('organization_id');
+    const federatedGraphId = c.req.param('federated_graph_id');
+    // Check authentication
+    if (
+      organizationId !== c.get('authenticatedOrganizationId') ||
+      federatedGraphId !== c.get('authenticatedFederatedGraphId')
+    ) {
+      return c.text('Forbidden', 403);
+    }
+    const key = `${organizationId}/${federatedGraphId}/routerConfigs/latest.json`;
+    let configStream: ReadableStream;
+    try {
+      configStream = await storage.getObject(c, key, 'no-cache');
+    } catch (e: any) {
+      if (e instanceof Error && e.constructor.name === 'BlobNotFoundError') {
+        return c.notFound();
+      }
+      throw e;
+    }
+    c.header('Cache-Control', 'private, no-cache, no-store, max-age=0, must-revalidate');
+    c.header('Content-Type', 'application/json; charset=utf-8');
+
+    const teedStream = configStream.tee();
+
+    const routerConfig = await streamToJSON(teedStream[0]);
+    const body = await c.req.json();
```

Comment:
Please validate "body.version"

Reply:
you mean validate that its a uuid?

Reply:
That the parameter is set and return 400 if not.

Reply:
Done